src/main/python/snakepit/main.py

```diff
@@ -84,6 +84,43 @@ def custom_output_filename(filename, output_directory):
     return osp.join(output_directory, filename)
 
 
+def build_template(distribution, yaml_spec):
+    if distribution == "miniconda":
+        template_filename = 'TEMPLATE-miniconda.spec'
+    elif distribution == "pyrun":
+        template_filename = 'TEMPLATE-PyRun.spec'
+        template_library_extension = 'TEMPLATE-PyRun-libraries-extension.ext'
+
+    # load the template
+    template = Environment(
+        loader=PackageLoader('snakepit', 'templates')
+    ).get_template(template_filename)
+
+    if distribution == "pyrun" and yaml_spec.get('libraries'):
+        template_extension = Environment(
+            loader=PackageLoader('snakepit', 'templates')
+        ).get_template(template_library_extension)
+        template = Environment().join_path(template_extension, template)
+
+    # render the template
+    return template.render(**yaml_spec)
+
+
+def construct_build_number(distribution, build, yaml_spec):
+    # create the build number
+    yaml_spec['build'] = build
+    if distribution == "miniconda":
+        build_number = "{0}_{1}{2}_{3}".format(yaml_spec['build'],
+                                               yaml_spec['conda_dist_flavour'],
+                                               yaml_spec['conda_dist_flavour_version'],
+                                               yaml_spec['conda_dist_version'],)
+    elif distribution == "pyrun":
+        build_number = "{0}_pyrun_{1}".format(yaml_spec['build'],
+                                              yaml_spec['pyrun_dist_version'])
```

Comment:
I would prefer to add `_py2.7` to the Release. I would put it in the end because it is least relevant but building with a higher Python version will still give you an upgrade path.